hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/s3/security/TestS3GetSecretRequest.java

```diff
@@ -329,11 +334,48 @@ public void testGetSecretOfAnotherUserAsNonAdmin() throws IOException {
         + "the permission to get bob's secret in this test case!");
   }
 
+  @Test
+  public void testGetSecretOfAnotherUserAsOzoneAdmin() throws IOException {
+    // This effectively makes alice an admin.
+    when(ozoneManager.isS3Admin(ugiAlice)).thenReturn(true);
+
+
+    int txLogIndex = 1;
+
+    S3GetSecretRequest s3GetSecretRequest = new S3GetSecretRequest(
+        new S3GetSecretRequest(
+            s3GetSecretRequest(USER_CAROL)
+        ).preExecute(ozoneManager));
+
+    // Run validateAndUpdateCache
+    OMClientResponse omClientResponse =
+            s3GetSecretRequest.validateAndUpdateCache(ozoneManager,
+                    txLogIndex, ozoneManagerDoubleBufferHelper);
+
+    // Check response type and cast
+    Assert.assertTrue(omClientResponse instanceof S3GetSecretResponse);
+    final S3GetSecretResponse s3GetSecretResponse =
+        (S3GetSecretResponse) omClientResponse;
+
+    // Check response
+    final S3SecretValue s3SecretValue = s3GetSecretResponse.getS3SecretValue();
+    Assert.assertEquals(USER_CAROL, s3SecretValue.getKerberosID());
+    final String awsSecret1 = s3SecretValue.getAwsSecret();
+    Assert.assertNotNull(awsSecret1);
+
+    final GetS3SecretResponse getS3SecretResponse =
+        s3GetSecretResponse.getOMResponse().getGetS3SecretResponse();
+    // The secret inside should be the same.
+    final S3Secret s3Secret1 = getS3SecretResponse.getS3Secret();
+    Assert.assertEquals(USER_CAROL, s3Secret1.getKerberosID());
+    Assert.assertEquals(awsSecret1, s3Secret1.getAwsSecret());
+  }
+
```

Comment:
Can we add a negative test case where e.g. `alice` is an Ozone admin, but **not** an S3 admin, thus can't do S3 secret operations (rejected) for others?